include/earcut.hpp

```diff
@@ -124,15 +124,15 @@ void Earcut<N>::operator()(const Polygon& points) {
         do {
             x = p->x;
             y = p->y;
-            if (x < minX) minX = x;
-            if (y < minY) minY = y;
-            if (x > maxX) maxX = x;
-            if (y > maxY) maxY = y;
+            minX = (std::min)(minX, x);
```

Comment:
The parentheses are on purpose. Otherwise it conflicts with <Windows.h> which is defines min and max with preprocessor-macros.
Otherwise whoever is including earcut in a windows project has to manually undefine and define those macros before including earcut.hpp, which is not a good idea imo.